coursera/cookies.py

```diff
@@ -108,6 +115,39 @@ def prepare_auth_headers(session, include_cauth=False):
     return headers
 
 
+def login_with_webview():
+    """
+    Creates a physical login window for the use to login.
+    """
+    url = "https://www.coursera.org/?authMode=login"
+
+    def get_cookies():
+        timeout = 120
+
+        def is_logged_in():
+            return webview.evaluate_js("document.querySelector('#logout-btn') !== null")
+
+        while not webview.webview_ready() or not is_logged_in():
```

Comment:
Is it possible to detect that the window has been closed and exit early? I tried closing the window and the script just continued to wait, but it should report that the user has closed the window and did not log in.